### PR TITLE
Optimize bottom_content query

### PR DESCRIPTION
**Pull request by benhalpern**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [x] Optimization
- [ ] Documentation Update

## Description

This PR optimizes the bottom content which shows up under articles by using the article's `cached_user` instead of including the user. This should improve performance and memory.

It also extends the lifespan of the cache because I don't think this content needs to be overly fresh and it is not cleared often.

I also adjusted the offset because I think this could be a chance to recycle slightly more classic articles and expose more articles to re-indexing via Google. Also upped the number displayed from 4 to 10. Since these articles are the last thing on the page after a bunch of other content, I think this can be an area to serve bot crawling as much as anything.

Screen Shot 2020-02-07 at 1 26 37 PM

### CODE REVIEW

File: app/labor/article_suggester.rb [hunk 1]
```diff
@@ -40,12 +39,11 @@ def suggestions_by_tag(max: 4)
...
       where.not(id: article.id).
       order("hotness_score DESC").
-      includes(user: [:pro_membership]).
       offset(rand(0..offsets[0])).
       first(max)
   end
 
   def offsets
-    Rails.env.production? ? [10, 120] : [0, 0]
+    Rails.env.production? ? [150, 1000] : [0, 0]
```
Comment (rhymes) [inline app/labor/article_suggester.rb]: it'd be great if we could document the reason behind these numbers, as you said in the PR, this way the thought process will stay in the code

Comment (rhymes) [reply app/labor/article_suggester.rb]: Also, is there a reason why we don't just return the two numbers and we differentiate between production and everything else? Both in development and testing 150 and 1000 items are very unlikely to be in the DB (which wouldn't matter in development anyway) and in the worst case scenario this only slows down a few tests.

File: app/labor/article_suggester.rb [hunk 2]
```diff
@@ -40,12 +39,11 @@ def suggestions_by_tag(max: 4)
     Article.published.tagged_with(article.tag_list, any: true).
       where.not(id: article.id).
       order("hotness_score DESC").
-      includes(user: [:pro_membership]).
```
Comment (mstruve) [inline app/labor/article_suggester.rb:43]: 🔥

File: app/views/articles/show.html.erb [hunk 3]
```diff
@@ -202,7 +202,7 @@
                  follow_cue: "Follow <a href='#{@classic_article.user.path}'>@#{@classic_article.user.username}</a> to see more of their posts in your feed." %>
     <% end %>
     <div id="additional-content-area" data-article-id="<%= @article.id %>,<%= @classic_article&.id %>"></div>
-    <% suggested_articles = ArticleSuggester.new(@article).articles(max: 4) %>
+    <% suggested_articles = ArticleSuggester.new(@article).articles(max: 10) %>
```
Comment (mstruve) [inline app/views/articles/show.html.erb:205]: This feels a little heavy to me but I like your reasoning for including more so Im game for whichever you want to try.

File: app/labor/article_suggester.rb (hunk 1 above)
Comment (nickytonline) [reply app/labor/article_suggester.rb:47]: Do we do a lot of ifs check whether it's prod or not? Wouldn't this make more sense in the site configuration?

**Comment by mstruve** [comment]
@benhalpern I got these specs passing and approved it, @nickytonline had one question. Is this something you still want to get out or table?

**Comment by mstruve** [comment]
Closing this for now, we can reopen if we decide we want it!

**Pull request closed without merging**